### PR TITLE
test(playwright): more flaky tests

### PR DESCRIPTION
**Pull request by stephenogorman**
### Proposed behaviour

Fix flaky Batch-Selection, Button & Dialog Playwright tests

### Current behaviour

Tests regularly fail during CI checks

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Additional context

N/A

### Testing instructions
- [x] Run Button tests in Playwright Test Runner to check if the `button.pw.tsx` file passed
- [x] Run Batch-Selection tests in Playwright Test Runner to check if the `batch-selection.pw.tsx` file passed
- [x] Run Dialog tests in Playwright Test Runner to check if the `dialog.pw.tsx` file passed
- [x] Run all other tests in Playwright Test Runner to check none of the other `*.pw.tsx` files have regressed

### CODE REVIEW

File: src/components/batch-selection/batch-selection.pw.tsx [hunk 1]
```diff
@@ -39,43 +40,38 @@ test.describe("check BatchSelection component properties", () => {
   test("should check hidden BatchSelection", async ({ mount, page }) => {
     await mount(<BatchSelectionComponent hidden />);
     const batchSelection = batchSelectionComponent(page);
-    await expect(batchSelection.getAttribute("hidden")).not.toBeNull();
-
+    // await expect(batchSelection.getAttribute("hidden")).not.toBeNull();
```
Comment (divyajindel) [inline src/components/batch-selection/batch-selection.pw.tsx:43]: Please remove this commented code.

**Review by divyajindel — CHANGES_REQUESTED** [review]

File: src/components/dialog/dialog.pw.tsx [hunk 2]
```diff
@@ -270,6 +270,7 @@ test.describe("Testing Dialog component properties", () => {
     const firstTextbox = page.getByLabel("Textbox1");
     const closeButton = page.getByLabel("Close");
 
+    await page.waitForTimeout(1000);
```
Comment (sianford) [inline src/components/dialog/dialog.pw.tsx:273]: Why are these timeouts needed? 1 second seems like a long time.

Comment (stephenogorman) [reply src/components/dialog/dialog.pw.tsx]: Each of the dialog tests that either tabs or shift+tabs through the elements regularly fails without some kind of wait before the first tab, especially when running through the UI. I'm not a fan of putting in arbitrary wait times, but this at least makes the test do what it's supposed to. I've reduced each wait time to a quarter of a second.

**Review by divyajindel — APPROVED** [review]

**Review by sianford — APPROVED** [review]

**Pull request merged**